### ISSUE TITLE
Sharing DAB purchases on specific social media channels

### ISSUE THREAD

**Issue by tbeck** [ocupop/comission-shopify-hydrogen#21]
**From Bill:**
I was wondering if everything has been thought through and taken in to account in terms of linking our product with users' social media platforms? Initially the requirement will be pretty simple as they will have just their certificate or DAB to post. Is that going to be a straight forward thing for them to do from their various devices?

**Comment by tbeck** [comment]
@ZiggyAtl - Can we confirm social media channels? I am assuming that it would be Twitter and Facebook to begin with?

@jennyvanderheiden - Can you adjust the order detail page design to accommodate for social icons and a "share" message?
https://xd.adobe.com/view/c43d3fbc-1701-492c-b37e-e66f529f8fa8-916e/screen/39d5a3f3-f649-43f3-bfb3-a3142f118473/

**Comment by jennyvanderheiden** [comment]
I have updated the layout to include that on the link above - will work on a social share image next

**Comment by ZiggyAtl** [comment]
Social would be Twitter, Facebook and Instagram. Instagram is by far the most important.

**Comment by paulstroot** [comment]
This is in progress and I have an early iteration up on the staging site. Once a DAB is purchased, the user's account page shows social media links for Facebook, Twitter and Instagram next to each dab in their order history. We will need to investigate a little further how specifically we would like to share for each social media platform, as there are some limitations to each one.

![image](https://user-images.githubusercontent.com/52838369/190265599-3735ffda-1508-4fa9-9068-c797a1b5ee9d.png)

**Comment by ZiggyAtl** [comment]
Thanks for the update Paul. Are you able to show me exactly what it will look like when it's shared, or is that what you were referring to in your last msg?

**Comment by tbeck** [comment]
@paulstroot - can you update the Facebook share to ensure that it uses the popup feature?

I think it is something like:
`?display=popup`

**Comment by ZiggyAtl** [comment]
For the social media posts, I highly doubt we'll be allowed by the copyright owner to post the entire digital version of the artwork. This was another reason we wanted the certificate as people could post those instead. And because the image on the certificate has the cross hair on there it protects the overall integrity of the original artwork (ie if someone copies it has a cross hair on there). So we need the certificate to be the thing which links with social media.

Twitter - just tweets the wording and doesn't attached the graphic. Can't it add the graphic (certificate) too?
Will the wording it currently creates for the tweet be cast in stone, or is this something we can modify easily down the line? When I clicked on the link that was contained in the tweet it went to a "Soemthings wrong here" page.

Instagram - Is it not possible for the graphic image to go straight in to a post, rather than just having a window telling people how to create the post? The link to download the image doesn't work.

Facebook - again, it would need to post the certificate. We need the ability for someone to enter the wording of the post to accompany the graphic before it gets posted (to save them then having to go to the post to add wording)

**Comment by tbeck** [comment]
Hi @ZiggyAtl - Thanks for the feedback and questions. We can step through the options for sharing and any limitations imposed by each platform during today's call. I think that we need to confirm what details of the certificate should be publicly accessible via social sharing so that we can adjust our meta tags to include these details in both the Facebook and Twitter sharing links.  

RE: Instagram... @paulstroot correct me if I am wrong here.
Instagram does not offer Share or Like buttons—which makes sense, since the nature of Instagram as a mobile photo- and video-sharing platform means it’s not really suited to liking and sharing web content. This means that the user will need to download the image asset we provide and create an Instagram post manually. We can adjust this download to link directly to the watermarked (crosshair) image as well though.

**Comment by paulstroot** [comment]
Tom. Correct about instagram not offering "Share" or "Like" buttons. I think our solution of giving instructions and providing an image to post is a good solution.

FYI, regarding social media sharing, I've made good progress on creating a dynamic page for users to share their DAB purchase directly on Facebook and Twitter. We need to work a bit on the design of this page (@jennyvanderheiden) , and the exact wording to use for the title and description. @alexjlahr , we'll be hitting you up for some advice on wording here. Feel free to reach out on slack for details.

**Comment by paulstroot** [comment]
> @paulstroot - can you update the Facebook share to ensure that it uses the popup feature?
> 
> I think it is something like: `?display=popup`

Now opens up in a popup window

**Comment by ZiggyAtl** [comment]
@paulstroot I believe you still need the suggested wording to use for the social shares? If so, I put this together, although happy for @Alexjlahr to suggest any improvements;
Check out the piece of [ARTWORK NAME] by [ARTIST] I’ve just bought at CoMission.Art. DAB# [DAB#] now has my name on it and [MUSEUM NAME] has a donation in my name

Is it possible that the message can be slightly different per social channel. The above message works ok for FB & IG as they are both @CoMission.Art however Twitter is @CoMission_art

Also when I tried the FB share the graphic in the post doesn't link to anything. Assume you are still working on that?

**Comment by ZiggyAtl** [comment]
@paulstroot also noticed a couple of things in the window which pops up for the instagram share. It says 'Download the image here' but there's no hyperlink on the 'here', and also there's a typo in the 3rd bullet point (r in about)

**Comment by ZiggyAtl** [comment]
@paulstroot I spoke with Tom yesterday about the first item below and he OK'ed it.

- [ ] Add three more free type boxes in the Museum Admin set up page for Instagram Handle, Twitter Handle and Facebook Username. In the Instagram and Twitter boxes please can it be pre-populated with the @ sign. In the Facebook one please can pre-populate it with www.facebook.com/YourUserName 
In the ? info box beside each one please can you put;
For IG - This is your unique Instagram username which starts with @ 
For Twitter - This is your unique Twitter username which starts with @ and appears in your profile URL
For Facebook - To find your username, click your profile picture in the top right of Facebook. Select Settings & privacy, then click Settings and you will be able to see your username

- [ ] Instagram:
Please change example copy to;
Check out the piece of [ARTWORK NAME] by [ARTIST] I’ve just bought @CoMission.Art. DAB# [DAB#] now has my name on it and [MUSEUM IG HANDLE] has a donation in my name

Twitter:
The image which gets posted has a reference to a SEO description (see screen shot)
Please use this wording for the post;
Check out the piece of [ARTWORK NAME] by [ARTIST] I’ve just bought @CoMission_Art. DAB# [DAB#] now has my name on it and [MUSEUM TWITTER HANDLE] has a donation in my name [LINK TO ARTWORK AT COMISSION.ART]

**Comment by paulstroot** [comment]
These items should be all set and ready for review. You can test the social media links by going to any existing purchases in your "Account" page.

**Comment by ZiggyAtl** [comment]
@paulstroot The names of the boxes in the edit screen of the museum app need changing please, as the facebook one requires a @ (which is shouldn't) and the IG one doesn't (which it should). Therefore if change the name of the second box to "Instagram Handle" and the third box to "Facebook Username" that will correct the issue. Also once they have been approved, the titles of the boxes don't match what is in the edit screen. Please can you make them consistent with the edit screen 

![Screen Shot 2022-11-09 at 2 31 20 PM](https://user-images.githubusercontent.com/112654923/200924034-e4625a37-6307-4915-81dc-dd9aea69cb9c.png)

![Screen Shot 2022-11-09 at 2 35 17 PM](https://user-images.githubusercontent.com/112654923/200924581-6d748756-aed7-4210-949a-a47ff5adc63c.png)

**Comment by paulstroot** [comment]
@ZiggyAtl Update made.

**Comment by ZiggyAtl** [comment]
Hi @paulstroot everything is working correctly in terms of the auto messages and the changes to the museum set up, thanks. The only thing is that the links to the certificate which appear in the suggested messages don't work, and I got this message below. I'm also having problems accessing a DAB I have just bought so I'll let you know if thats just my user error or whether it might be a bug

![Screen Shot 2022-11-16 at 11 34 52 AM](https://user-images.githubusercontent.com/112654923/202239210-16fe4ffc-9bda-412c-9316-e340814f59f3.png)

**Comment by paulstroot** [comment]
I just noticed this one too. I'll get that taken care of today.

**Comment by paulstroot** [comment]
HI @ZiggyAtl . We've been working hard to track down an issue that was causing many of these issues with the certificates and social media images being generated and we believe we've worked it out today. I just pushed an update that should fix these things. I will be submitting some test orders today to test it out. Please feel free to test on your end as well to make sure it's working properly.

**Comment by ZiggyAtl** [comment]
@paulstroot I had thought this was fixed but when I try posting it through facebook (via Safari), the image doesn't appear on the post, see below
![Screen Shot 2022-11-28 at 9 22 49 PM](https://user-images.githubusercontent.com/112654923/204423246-56f4f7a3-9597-41d8-adca-0b42b14b7a42.png)

**Comment by paulstroot** [comment]
@ZiggyAtl , I'lll look into that today. I have a couple follow-up questions:

1. Are you able to find out exactly which piece of art and dab number you tried to share? I'll try it myself also to see if I can recreate the issue.

2. When you click the facebook share button, and the popup appeared where you can type in your custom message, did the correct title and description display? See my screenshot below as an example:

![image](https://user-images.githubusercontent.com/52838369/204578025-bbf52888-467b-407c-8241-a4d2a57aafe9.png)

_(Note: I've noticed that the very first time you share a piece of art the image does not show up on this page. Subsequent tiles it will. This is because Facebook retrieves these images the first time an image is shared, so it isn't available to display in this popup until the second time the image is shared)_

**Comment by ZiggyAtl** [comment]
@paulstroot heads up I'm going to do some testing on this now. I also had the same problem the other night with a twitter one, and like a fool I forget to note which one it was. Will get back to you shortly

**Comment by ZiggyAtl** [comment]
@paulstroot I just went through the process again and bought another dab (https://store.comission.art/dab/7273842147507-175) but when I clicked on the link in the email I got, this came up

![Screen Shot 2022-12-08 at 2 58 39 PM](https://user-images.githubusercontent.com/112654923/206555811-84c8ebda-bbe1-4ad1-bda1-efd5919b710a.png)

**Comment by paulstroot** [comment]
Got it! Good catch Bill. I'm uploading a fix for that right now. Here is
the working URL

https://store.comission.art/dab/7273842147507-175

**Comment by ZiggyAtl** [comment]
@paulstroot I've tried it a couple of times since (Fri and today) and it still doesn't seem to be working. I just purchased a DAB (Heels Out #810) through my bill@soccerstreets.org account, and clicked on the View Your Certificate link in the welcome email. It took me to this page, as it should
![Screen Shot 2022-12-13 at 12 59 38 PM](https://user-images.githubusercontent.com/112654923/207410319-b5be0d59-f64b-48eb-89dc-305d087b9481.png)

But then when I clicked on the download certificate and then tried to open the certificate I got this screen;
![Screen Shot 2022-12-13 at 1 01 53 PM](https://user-images.githubusercontent.com/112654923/207410459-aebfeb6f-36c5-4fb0-958c-8d4c2cdf6897.png)

Also from that previous screen I hit the link to post it to twitter, but the link the system put in the tweet also took me to the Somethings wrong here page. The same also happened with the post i made to Facebook.

I then accessed it through my account, not through the link in the welcome email, and when I clicked view certificate it opened, however when I clicked the Download Certificate button the file also took me to the Something Wrong page. And the social media posts also pointed to that page too.

**Comment by paulstroot** [comment]
Got it Bill. I'll put this next on my list to investigate!

**Comment by paulstroot** [comment]
I just uploaded an update. Give it a shot now.

**Comment by ZiggyAtl** [comment]
@paulstroot that seems to have fixed it. I've just realized it wasn't sending an image to use in the tweet, and this has been corrected now too. I only tested it with a one DAB purchase. Is this worth testing with a two DAB purchase, or if it's fixed for one will it automatically be OK for two?

**Comment by paulstroot** [comment]
It should be fixed for all DAB purchases going forward. That said though,
it's never a bad idea to continue to test :)

**Comment by paulstroot** [comment]
I think what happened is that the purchase for this DAB was made before I
made my last fix to the site a couple days ago. Purchases before that had
an issue generating that Instagram image.

I just went through and re-generated the instagram image for the couple
older purchases where I noticed it was missing. I do believe all purchases
of "Heels Out" and "This Illustrious Feeling" should have all imagery
generated now.
